src/main/scala/com/fulcrumgenomics/rnaseq/CollectErccMetrics.scala

```diff
@@ -0,0 +1,294 @@
+      }
+      .foreach { r =>
+        erccReadPairs += 1
+        erccReadPairCounter.count(r.refName)
+      }
+    in.safelyClose()
+
+    // Transform to log2 counts to compute some simple correlation and linear regression metrics.
+    val counts            = ArrayBuffer[Double]()
+    val normalizedCounts  = ArrayBuffer[Double]()
+    val concentrations    = ArrayBuffer[Double]()
+    val maximumErccLength = erccData.keys.map { name => in.header.getSequence(name).getSequenceLength }.max
+    erccData.iterator.foreach { case (name, concentration) =>
+      val count = erccReadPairCounter.countOf(name)
+      if (minTranscriptCount <= count) {
+        val erccLength = in.header.getSequence(name).getSequenceLength
+        counts += log2(count)
+        normalizedCounts += log2(count * maximumErccLength / erccLength.toDouble) // normalize by the ERCC transcript length
+        concentrations += log2(concentration)
+      }
+    }
+
+    // Rocket science
+    val simpleRegression = new SimpleRegression()
+    normalizedCounts.zip(concentrations).foreach { case (cnt, conc) => simpleRegression.addData(cnt, conc) }
+
+    val erccMetrics = ErccSummaryMetrics(
+      total_reads                = totalReads,
+      ercc_reads                 = erccReads,
+      fraction_ercc_reads        = if (totalReads == 0) 0 else 100.0 * erccReads / totalReads.toDouble,
+      ercc_templates             = erccReadPairs,
+      total_transcripts          = erccReadPairCounter.iterator.length,
+      total_filtered_transcripts = erccReadPairCounter.iterator.count(_._2 >= minTranscriptCount),
+      pearsons_correlation       = noneIfNaNOrInsufficientData { new PearsonsCorrelation().correlation(normalizedCounts.toArray, concentrations.toArray) },
+      spearmans_correlation      = noneIfNaNOrInsufficientData { new SpearmansCorrelation().correlation(normalizedCounts.toArray, concentrations.toArray) },
+      intercept                  = noneIfNaN { simpleRegression.getIntercept },
+      slope                      = noneIfNaN { simpleRegression.getSlope },
+      r_square                   = noneIfNaN { simpleRegression.getRSquare }
+    )
+
+    // Write the output
+    {
+      def f(ext: String): FilePath = PathUtil.pathTo(output + ext)
+
+      val summaryPath  = f(".ercc_summary_metrics.txt")
+      val detailedPath = f(".ercc_detailed_metrics.txt")
+      val plotPath     = f(".ercc_plot.pdf")
+
+      // summary metrics file
+      Metric.write[ErccSummaryMetrics](summaryPath, erccMetrics)
+
+      // detailed (per-ERCC-transcript) file, written in the order defined in the Sam header
+      val metrics = in.header.getSequenceDictionary
+        .getSequences
+        .map(_.getSequenceName)
+        .filter { name => erccData.contains(name) }
+        .map { name =>
+          val count           = erccReadPairCounter.countOf(name)
+          val normalizedCount = count / in.header.getSequence(name).getSequenceLength.toDouble
```

Comment:
I think it's a little odd that you normalize in two different ways.  Above it's by `maxTxLen / len` and here's it just be `len`.  They're equivalent since `maxTxLen` is a constant, but it seems strange to do it differently in two places.